sonar-plugin/javascript-checks/src/main/java/org/sonar/javascript/checks/HardcodedSecretsCheck.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.javascript.checks;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.plugins.javascript.api.Check;
+import org.sonar.plugins.javascript.api.JavaScriptRule;
+import org.sonar.plugins.javascript.api.TypeScriptRule;
+
+@TypeScriptRule
+@JavaScriptRule
+@Rule(key = "S6418")
+public class HardcodedSecretsCheck extends Check {
+
+  private static final String DEFAULT_SECRET_WORDS = "api[_.-]?key,auth,credential,secret,token";
+  private static final String DEFAULT_RANDOMNESS_SENSIBILITY = "3.0";
+
+  @RuleProperty(
+    key = "secret-words",
+    description = "Comma separated list of words identifying potential secrets",
+    defaultValue = DEFAULT_SECRET_WORDS
+  )
+  public String secretWords = DEFAULT_SECRET_WORDS;
+  @RuleProperty(
+    key = "randomness-sensibility",
+    description = "Minimum shannon entropy threshold of the secret",
+    defaultValue = DEFAULT_RANDOMNESS_SENSIBILITY
+  )
+  public String randomnessSensibility = DEFAULT_RANDOMNESS_SENSIBILITY;
+
+  @Override
+  public List<Object> configurations() {
+    return Collections.singletonList(
+      new HardcodedSecretsCheck.Config(secretWords, randomnessSensibility)
```

Comment:
You don't really need to prefix the inner class with the outer class because the inner class is static (Java FTW).

```suggestion
      new Config(secretWords, randomnessSensibility)
```